### PR TITLE
docs: api/grn_table: move grn_table_difference() reference to the header file

### PR DESCRIPTION
**Pull request by abetomo**
GitHub: GH-1817

Prepare to switch to documents automatically generated by Doxygen.

### CODE REVIEW

File: include/groonga/table.h [hunk 1]
```diff
@@ -709,6 +709,18 @@ grn_table_setoperation(grn_ctx *ctx,
...
+ * \param table1 The table.
+ * \param table2 The table.
+ * \param res1 Specify `table1`.
+ * \param res2 Specify `table2`.
+ *
+ * \return \ref GRN_SUCCESS on success, \ref GRN_INVALID_ARGUMENT on
+ *         `table1 != res1 || table2 != res2`.
+ */
 GRN_API grn_rc
 grn_table_difference(
```
Comment (otegami) [inline include/groonga/table.h:725]: Is the grn_table_difference function no longer in use? If that's the case, should we consider adding a deprecation notice?

File: include/groonga/table.h [hunk 2]
```diff
@@ -709,6 +709,18 @@ grn_table_setoperation(grn_ctx *ctx,
                        grn_obj *table2,
                        grn_obj *res,
                        grn_operator op);
+/**
+ * \brief Delete duplicate records in `table1` and `table2`.
```
Comment (kou) [inline include/groonga/table.h:713]:
```suggestion
 * \brief Delete duplicated records in `table1` and `table2`.
```

File: include/groonga/table.h (hunk 1 above)
Comment (kou) [reply include/groonga/table.h:725]: We don't need to do it so far.

We may want to do it when we add `GRN_OP_DIFFERENCE` and add support for it in `grn_table_setoperation()`. (But in-place difference can't be done with `grn_table_setoperation()` interface...)

Comment (otegami) [reply include/groonga/table.h:725]: I got it, thank you!

**Pull request merged**